### PR TITLE
Chore: Convert TracePageHeader test to RTL

### LINKED ISSUE #51227: Convert `packages/jaeger-ui-components/src/TracePageHeader/TracePageHeader.test.js` to RTL

### PR DESCRIPTION
**Pull request by Seyaji**
**What this PR does / why we need it**:

Fixes https://github.com/grafana/grafana/issues/51227

### CODE REVIEW

**Comment by Seyaji** [comment]
@ashharrison90 can you create an issue for this?  🙂

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
couple of tweaks, otherwise lgtm 😄

File: packages/jaeger-ui-components/src/TracePageHeader/TracePageHeader.test.js [hunk 1]
```diff
@@ -12,75 +12,86 @@
...
-  it('renders an empty <div> if a trace is not present', () => {
-    wrapper = mount(<TracePageHeader {...defaultProps} trace={null} />);
-    expect(wrapper.children().length).toBe(0);
-  });
+  it('should render an empty <div> if a trace is not present', async () => {
+    setup({ trace: null });
 
-  it('renders the trace title', () => {
-    expect(wrapper.find({ traceName: getTraceName(trace.spans) })).toBeTruthy();
+    expect(await screen.findAllByRole(/[<div \/>]/)).toHaveLength(1);
```
[the remark this answers is not in the document]
Comment (Seyaji) [reply packages/jaeger-ui-components/src/TracePageHeader/TracePageHeader.test.js:52]: Agreed 👍

File: packages/jaeger-ui-components/src/TracePageHeader/TracePageHeader.test.js [hunk 2]
```diff
@@ -12,75 +12,120 @@
...
 
     it('hides the summary when hideSummary === true', () => {
-      expect(wrapper.find(LabeledList).length).toBe(1);
-      wrapper.setProps({ hideSummary: true });
-      expect(wrapper.find(LabeledList).length).toBe(0);
+      const { rerender } = setup({ hideSummary: false });
+      expect(screen.queryAllByRole('listitem')).toHaveLength(5);
+
+      rerender(<TracePageHeader hideSummary={false} trace={null} />);
+      expect(screen.queryAllByRole('listitem')).toHaveLength(0);
```
Comment (Seyaji) [inline packages/jaeger-ui-components/src/TracePageHeader/TracePageHeader.test.js:96]: Becuase I was rerendering the component inline without the full props defined in setup the trace was defaulting to null, so it wasnt rendering anything, I left this in as an example.

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**